src/Post/PostServiceProvider.php

```diff
@@ -9,11 +9,36 @@
 
 namespace Flarum\Post;
 
+use DateTime;
 use Flarum\Event\ConfigurePostTypes;
 use Flarum\Foundation\AbstractServiceProvider;
 
 class PostServiceProvider extends AbstractServiceProvider
 {
+    /**
+     * {@inheritdoc}
+     */
+    public function register()
+    {
+        $this->app->extend('flarum.api.floodgates', function ($floodgates) {
+            $floodgates['createPostFloodgate'] = [
+                'paths' => ['/api/posts', '/api/discussions'],
```

Comment:
i think this should use the route _names_ instead, as the paths could potentially change.

Reply:
That would require us to check that the route exists and map the path to a route, which only happens in DispatchRoutes. While I'd agree it'd be nicer, the extra boilerplate might not be worth it.

Reply:
I'm not talking about nicer. An extension could change it and then the floodgate would become useless.

Reply:
Then the same should probably be done for the CSRF extender. We'd probably also want to turn the path -> route retrieval logic into it's own system.

Reply:
Hm, perhaps.

Reply:
After looking more at the code, I'm not sure how feasible this is if we want to use middleware. We would effectively need to run DispatchRoutes twice. The best way that comes to my mind of doing this is:

- Run DispatchRoutes somewhere in the middle of the middleware stack.
- Instead of directly running the controller, save it as a request attribute, alongside params and the route name
- At the end of the middleware stack, run it through the controller.

The end result is a bit messier and less direct. It also loses the advantage of rate-limiting spammers who indiscriminately check every endpoint they can, meaning that the site could be probed for which endpoints do / do not exist moreso than now.

Reply:
Implemented in https://github.com/flarum/core/pull/2425.